### PR TITLE
extract_dataset return DataArray if single variable

### PR DESCRIPTION
**Pull request by aloctavodia**

### CODE REVIEW

**Comment by OriolAbril** [comment]
We could have an `extract_dataset` and an `extract_dataarray` function. Now seeing this I am worried about users being even more confused about xarray objects if they use `extract_dataset` and get a DataArray

**Comment by aloctavodia** [comment]
I may be using the function in the wrong way, but in general if I ask for 

`ext = az.extract_dataset(idata, var_names="mu")`

It is because I want to operate directly over mu, so I end up doing something like  `ext["mu"]` which is redundant (and annoying). An alternative is to do  `az.extract_dataset(idata)["mu]`, but that's just a workaround.

So I think this should be the default behavior. We can have an argument, if users want to keep a Dataset even for a single variable. And we can change the name of the function to avoid confusions. Having two functions for so similar operations seems too much complexity.

File: arviz/data/utils.py [hunk 1]
```diff
@@ -14,7 +15,37 @@ def extract_dataset(
     num_samples=None,
     rng=None,
 ):
-    """Extract an InferenceData group or subset of it as a :class:`xarray.Dataset`.
+    """Extract an InferenceData group or subset of it.
+
+    This function has been deprecated please use extract
```
[the remark this answers is not in the document]
Comment (OriolAbril) [reply arviz/data/utils.py:20]: 0.12.1 eas released already qnd didn't have the deprecation. The first release with the deprecation will be 0.13. We can then remove it in 0.14 or 0.15 (we should probably create an issue and/or milestoneto remember aboyt this)

[the remark this answers is not in the document]
Comment (aloctavodia) [reply arviz/data/utils.py]: And do we have to change this if we make a  0.12.x release?

**Pull request merged**